pytorch_lightning/core/lightning.py

```diff
@@ -1447,6 +1447,21 @@ def optimizer_step(
             ``training_step()``, ``optimizer.zero_grad()``, ``backward()`` are called within
             :meth:`~pytorch_lightning.trainer.training_loop.TrainLoop.run_training_batch`.
 
+        Warning:
+            When using the standard trainer and ``accumulate_grad_batches``, both
+            ``optimizer.step()`` and ``optimizer.zero_grad()`` are called only once per
+            *optimization cycle*, and not once per step. This means that code like
+            the following:
```

Comment:
I am not convinced about adding this warning. Optimizer step not getting called during the accumulation phase is the expected behavior since this is really just what accumulation means, i.e., no stepping! 

A simple line like "This method won't be called during the accumulation phase when Trainer(accumulate_grad_batches != 1)" would totally do imo.

Reply:
I agree - I was following the conversation here: https://github.com/PyTorchLightning/pytorch-lightning/pull/4984#discussion_r537302856, and this was until recently an error, so I erred on the side of being particularly cautious. But this really is just expected behavior.